keiko/keiko/keiko.py

```diff
@@ -22,6 +22,35 @@
 
 DATA_SHAPE_CLASS_NAME = "DataShape"
 
+_latex_special_chars = {
+    "&": r"\&",
+    "%": r"\%",
+    "$": r"\$",
+    "#": r"\#",
+    "_": r"\_",
+    "{": r"\{",
+    "}": r"\}",
+    "~": r"\textasciitilde{}",
+    "^": r"\^{}",
+    "\\": r"\textbackslash{}",
+    "\n": "\\newline%\n",
+    "-": r"{-}",
+    "\xA0": "~",  # Non-breaking space
+    "[": r"{[}",
+    "]": r"{]}",
+}
+
+
+def latex_escape(s):
+    """Escape characters that are special in LaTeX.
+
+    References:
+    - https://github.com/JelteF/PyLaTeX/blob/ecc1e6e339a5a7be958c328403517cd547873d7e/pylatex/utils.py#L68-L100
+    - http://tex.stackexchange.com/a/34586/43228
+    - http://stackoverflow.com/a/16264094/2570866
+    """
+    return "".join(_latex_special_chars.get(c, c) for c in str(s))
```

Comment:
Please consider `str.maketrans` and `str.translate` instead. I believe it fits exactly for this purpose and it might be a little faster